### DISCUSSION TITLE
cookie-based color mode with nextjs 13

### DISCUSSION THREAD

**Discussion by bnn1** [vercel/next.js#49793] [Help]
### Summary

Hi guys. I implemented a cookie-based color mode for my web app. I'm reading cookies in my root layout and applying a class on the root element based on the cookie. I created a color mod toggle component - a button wrapped in `form` with server action. 
The problem I have with this approach is that it makes the whole application dynamically rendered because root layout reads the cookie. 
Is there a way to generate static pages inside dynamic root layout?

**Comment by wscourge** [comment]
@bnn1 how did you end up doing this?

**Reply by bnn1** [reply]
I went with client side color mode. AFAIK there's no way to use cookies without SSR

**Reply by wscourge** [reply]
OK, thanks for answering
